src/worker/utils/caching.ts

```diff
@@ -29,31 +29,29 @@ export const storeJSMolInCache = (smiles: string, mol: JSMol) => {
   const nbCachedMolecules = Object.keys(globalThis.rdkitWorkerGlobals.jsMolCache).length;
   if (nbCachedMolecules > globalThis.rdkitWorkerGlobals.maxJsMolsCached) {
     cleanJSMolCache();
-    globalThis.rdkitWorkerGlobals.jsMolCache = { [smiles]: mol };
-    return;
   }
   try {
-    globalThis.rdkitWorkerGlobals.jsMolCache[smiles] = mol;
+    globalThis.rdkitWorkerGlobals.jsMolCache.set(smiles, mol);
   } catch (e) {
     console.error(e);
     cleanJSMolCache();
-    globalThis.rdkitWorkerGlobals.jsMolCache = { [smiles]: mol };
+    globalThis.rdkitWorkerGlobals.jsMolCache.set(smiles, mol);
   }
 };
 
 export const getJSMolFromCache = (smiles: string) => {
   if (!globalThis.rdkitWorkerGlobals.jsMolCacheEnabled || !globalThis.rdkitWorkerGlobals.jsMolCache) {
     return null;
   }
-  return globalThis.rdkitWorkerGlobals.jsMolCache[smiles];
+  return globalThis.rdkitWorkerGlobals.jsMolCache.get(smiles);
 };
 
 export const cleanJSMolCache = () => {
   if (!globalThis.rdkitWorkerGlobals?.jsMolCache) return;
-  for (const [smiles, mol] of Object.entries(globalThis.rdkitWorkerGlobals.jsMolCache)) {
+  for (const [smiles, mol] of globalThis.rdkitWorkerGlobals.jsMolCache.entries()) {
     try {
       mol.delete();
-      delete globalThis.rdkitWorkerGlobals.jsMolCache[smiles];
+      globalThis.rdkitWorkerGlobals.jsMolCache.delete(smiles);
```

Comment:
- [ ] TODO: use `Map.clear `at the end of the loop, instead of n calls to `delete`

Reply:
I thought about something like this:

```ts
class MolCache extends Map {
  set(smiles) {
    this.set(smiles, rdkit.get_mol(smiles))
  }
  
  clear() {
    this.forEach((v, k) => {
      v.delete();
      this.delete(k); 
    )}
  }
}

// MolCache
cache.set('C')
cache.clear() //
```

Reply:
looks good, I'll do that